Global enable / disable, very basic FirePHP compatibility

ErikKrause opened the pull request:
Implemented a global enable / disable function and a setting to switch on or off FirePHP compatibility (defaults to off). If switched on argument order is changed if exactly 2 arguments are passed to the various log functions to be able to re-use old FirePHP calls.

ErikKrause committed:
Very basic FirePHP compatibility for ChromePhp

Added global enable / disable, added setting for FirePHP log style. If switched on argument order is changed such, that you can reuse old firephp calls.

ErikKrause committed:
Added Firefox hints

ErikKrause committed:
Added Firefox hints

ErikKrause committed:
Added Firefox hints

ErikKrause committed:
Added Firefox hints

ErikKrause committed:
Update README.md

ErikKrause committed:
Very basic FirePHP compatibility for ChromePhp

Added global enable / disable, added setting for FirePHP log style. If switched on argument order is changed such, that you can reuse old firephp calls.

ErikKrause committed:
Added Firefox hints

ErikKrause committed:
resources are loggable now.

json_encode failed for resources causing an empty log. Now uses print_r and get_resource_type for resource, print_r for all other.

DanMan commented:
Coming from FirePHP myself this is generally a good idea, but you changed a bit more than necessary, which I think is inappropriate. Maybe just add the switch and leave the rest up to the maintainer?

ghost commented:
Well, I changed exactly what was needed to make it work for me. Said that I'd like to know what exactly you think is more than necessary.

Asenar commented:
Hi @Erik-Krause / ( @erikkrause ?) thanks you very much for your adds !

Maybe you can change your composer.json to set the name as `erikkrause/composerphp` and register it in composer ?

Asenar committed:
Add help to use this repo in composer

ErikKrause committed:
Merge pull request #1 from Asenar/readme-help-composer

Add help to use this repo in composer

lorenzos committed:
Better logging of objects and non-string scalars

As suggested in [comments **here**](https://github.com/ErikKrause/chromephp/commit/e61698fdd5bda4163d2fd407d078bd9f8129cb20#commitcomment-21296454).

ErikKrause committed:
Merge pull request #2 from lorenzos/patch-1

Better logging of objects and non-string scalars

powtac committed:
Clearify documentation

ErikKrause committed:
Merge pull request #3 from powtac/patch-1

Clearify documentation

ErikKrause committed:
Update README.md

ErikKrause committed:
Update README.md

Jeremip11 commented:
Ty!

hopeseekr commented:
Whenever you fork a project via GitHub and you maintain its master branch way after you make a PR on the original PR, this is what happens. That's why I literally deleted my fork and just created a standalone repo.

hopeseekr commented:
@Erik-Krause Can you prepare a PR for phpexpertsinc/chromephp#3? I don't want the composer.json changes. I don't want all those `(ek)` comments, but your copyright clause comments are very good and well, especially since the Apache v2 license pretty much requires attribution.

ErikKrause commented:
What do you mean by "all those (ek) comments". I won't remove the FirePHP compatibility changes, since that's the whole point. Apart of that I like your proposal to move this to a new repo.
